### ISSUE TITLE
Some things I would like to change/fix

### ISSUE THREAD

**Issue by Alex313031** [ltguillaume/thorium-winupdater#1]
1. I would prefer the original "blue" logo be used instead of the greyscale one. We can still use the greyscale one other places, like maybe in a different part of the updater, and we can keep it in the readme.
2. I would prefer to redo the icons. They don't include all the sizes I want, and they are compressed, meaning they cant be shown on linux. I would like to re-create them the way I do all the other .ico files in the thorium repo.
3. Include building instructions, including info about autohotkey. Either as a ## Building section in the readme, or a seperate .md file.

**Comment by ltguillaume** [comment]
Yeah I was actually nearly done writing a response here when the smoke started 🙈

1. The reasons behind using a different color for WinUpdater is to 1) distinguish it from the browser in the taskbar when both are running, 2) making clear it's the updater running when the user sees its tray icon, not a part of the browser, or (in LibreWolf's case) the portable launcher, 3) to distinguish it from the browser if it's put in the Start menu (LibreWolf installer does this).

**Comment by Alex313031** [comment]
@ltguillaume 
Ok, 1. seems good, we can keep the greyscale icon.

2. I prefer to have 16x16, 32x32, 48x48, 64x64, 128x128, and 256x256 in each .ico, with only the 256x256 being compressed. Your .ico is missing some of these sizes. This is the default output of GIMP, which is what I use for creating icons for all the platforms including windows. The .ico files you created do not show up correctly on either native linux (in the file manager or photo viewer) or WINE. It is because you have all of the sizes compressed.

3. I made some modifications to my repo, I'm gonna send in a PR. I'm assuming you are using Ahk2exe.exe?

4. I'm kind of on the fence about this. Because it might be good for non tech savvy users. But at the same time, I think it's good to have manual control over this, if you wanted to change releases. Cr-Launcher by @henrypp uses this approach.

**Comment by ltguillaume** [comment]
2. https://github.com/Alex313031/thorium/blob/main/logos/NEW/win/thorium.ico is missing 128x128 😛 Ah ok, makes sense that they won't work under Linux then. Overall sound reasing.
3. Yeah I use Ahk2Exe (the v1 branch, v2 requires different syntax) and Resource Hacker to remove the unnecessary icons from the binary.
4. I'd say that would only make sense if WinUpdater would also download the browser the first time (without the user ever having downloaded a release at all). Currently, I think users who'd want to switch to another build would just download 't and overwrite the old one, while not relying on an updater to do the switch for them.

**Comment by ltguillaume** [comment]
By the way, when my PC caught fire (apart from the harddrive and the molex-to-SATA adapter, everything has survived without a scratch, remarkable!) I wanted to proposy this alternative icon, see what you thought of it.

![Thorium-WinUpdaterPurple](https://github.com/ltguillaume/thorium-winupdater/assets/4777345/09ef1266-ef75-4903-b95d-97b84f9b7f1a)
![SCREENSHOT](https://github.com/ltguillaume/thorium-winupdater/assets/4777345/0636e9c7-0634-4275-9fa7-141d66319ba3)

**Comment by Alex313031** [comment]
2. @ltguillaume Most .icos in the repo have 128x128. All .icos for any of my other projects have 128x128. The reason why thorium.ico doesn't have it, is because it is simply a renamed copy of the main one that is actually used in the Chromium/Thorium binary, here > https://github.com/Alex313031/thorium/blob/main/src/chrome/app/theme/chromium/win/chromium.ico

I wanted to match upstream to prevent issues. They have 16, 32, 48, 64, 256, with 256 coming last, rather than first, and only the 256 compressed. All .pngs at 72 DPI. So thats how I made the thorium icon, to match it.

3. Ahh ok, that would make sense why I couldn't build it with v2. It's kinda like Yarn 1 vs 2. Basically a totally different product.

4. Yeah, I can include a thor_ver file. The only thing, is how will the updater know to cd into the folder, since the version number will change with every release.

**Comment by ltguillaume** [comment]
2. I guess that's mostly right. So are we matching upstream (no 128x128)?
4. That isn't a problem, but why not put it next to `thorium.exe`, which isn't in the versioned folder anyway?

**Comment by Alex313031** [comment]
@ltguillaume 
2. I mean I could make a thorium.ico with 128x128 included, for use in this project. I only excluded it so that my icon matches the upstream Chromium one in every way, except for the actual content of the .pngs.
4. There are multiple problems with this. I cant move thorium.exe and chrome_

**Comment by Alex313031** [comment]
@ltguillaume Yeah. 

Thats genius, I will try this out!

**Comment by Alex313031** [comment]
@ltguillaume Are you gonna merge my PR?

**Comment by ltguillaume** [comment]
Yes I will. It's just that I have changes locally that still need some cleaning up, too, and I won't do a new release until there's a Thorium release with `thor_ver` included anyway.

**Comment by ltguillaume** [comment]
I'm afraid I don't know how to reliably translate this into 1) the proper repo and 2) the proper build from that repo, because of overlap:

1. `AVX` means from the `Thorium-Win` repo, so I'd rather see `Win` in the thor_ver file
1. `AVX2` means from the `Thorium-Win-AVX2` repo, so I'd rather see `Win-AVX2` in the thor_ver file
1. Now it gets tricky: in the `Thorium-Win7` repo, there is an AVX, AVX2, SSE3 and a WIN32 build. First, there's overlap between this repo's builds and the others (AVX, and AVX2). So I think it would make sense to have a thor_ver file here with on the first line `Win7` (the repo) and on the second line either `AVX`, `AVX2`, `SSE3` or `WIN32` (so the unique part in the filename).
1. Is there an SSE2 build? I don't see it anywhere

**Comment by Alex313031** [comment]
well I could make it go like
Win-AVX
Win-AVX2
Win-SSE3

and
Win7-AVX
Win7-AVX2
Win7-SSE3
Win7-SSE2

For the windows 7 builds, I also make special 32 bit SSE2 builds. This is in spite of Chromium requiring SSE3 since M89. I patched it to allow compiling. The sse2 builds are only in the win7 repo. They are designed for very old systems i.e. Pentium 4.

**Comment by ltguillaume** [comment]
- But the repo for Win-AVX is just called Win.
- I think the SSE2 build is then actually called WIN32

**Comment by Alex313031** [comment]
@ltguillaume Yeah, I can do this. Can you send examples of how you would like it. Because I don't really care how the files are, I wanna make it easiest for you.

**Comment by Alex313031** [comment]
@ltguillaume Another complication is that the SSE3 builds are hosted in Thorium-Special, which also has linux builds and used to host other OS builds.

**Comment by ltguillaume** [comment]
Hmz, I see. Well, as long as the latest release would always include a `....zip` and `...mini_installer.exe`, it wouldn't be a problem, because then a `thor_ver` file containing the following would suffice:
```
Special
SSE3
```

**Comment by Alex313031** [comment]
@ltguillaume I am releasing M117, so I am going to copy this https://github.com/ltguillaume/thorium-winupdater/issues/1#issuecomment-1709380222. Then we can test.

**Comment by ltguillaume** [comment]
Cool!

**Comment by Alex313031** [comment]
@ltguillaume https://github.com/Alex313031/thorium/commit/2194599894851f9e4787353416f667d67a03358d

I would like to start with just the non-win7 repos for now
Test and make sure everything works ok.

Today (October 10) Is actually the final day for the NT 6.x.x series. ESU ended today for Win7/8/8.1 and Server 2012/2012 R2. It is also today that Chromium released the final M109 version 109.0.5414.170, and Electron 22 released their final version 22.3.27.

I will be making a final build of Thorium 109 based on this version, and update my Electron apps to this version.

After that, I will be basing the win7 builds on [Supermium](https://github.com/win32ss/supermium) M118, and that is when I will add the thor_ver files to that repo.

**Comment by ltguillaume** [comment]
@Alex313031 See https://codeberg.org/ltguillaume/thorium-winupdater/releases/tag/1.8.0

I've tested by installing the previous version, but with the new `thor_ver` files (Win (AVX) and Win-AVX2):
1. System level installed version
2. Portable install

I tried to not (re)create icons or start the browser after updating (by using the respective command line parameters, as you suggested), but `--do-not-create-any-shortcuts` apparently doesn't work, so there's no way to not recreate Start menu and desktop icons... `--do-not-create-taskbar-shortcut` isn't around anymore either, but it doesn't seem the installer creates a taskbar icon anyway.

I have tested it via the scheduled task, too (and found a bit of bug in the mini installer I had to work around).

**Comment by ltguillaume** [comment]
With https://codeberg.org/ltguillaume/thorium-winupdater/releases/tag/1.8.1 everything seems to work fine in all conditions.

**Comment by Alex313031** [comment]
@ltguillaume Nice! Saw the changes you made. Does the updater now correctly get the repo it should download from based on the contents of the thor_ver file?

Also am publishing your releases here on Github > https://github.com/Alex313031/thorium-winupdater/releases/tag/1.8.1

**Comment by ltguillaume** [comment]
>Does the updater now correctly get the repo it should download from based on the contents of the thor_ver file?

Well yeah, that was the whole idea 😋 (see changelog, commits and the previous posts here). It shows the repo (and build) during the update after the version number (and in the .ini file after updating).

Obviously I couldn't test this with the Win7 releases yet, but it's set up to work for that, too.
